### PR TITLE
Rewrite how devfs_rulesets are generated

### PR DESCRIPTION
**Pull request by skarekrow**
- Simplify plugin devfs ruleset generation
- Rewrite generate_devfs_ruleset

We now no longer touch the hosts /etc/devfs.rules, instead we create a dynamic ruleset that we clean up on jail destruction.

This ends up fixing some bugs where setting dhcp after setting allow_tun or vice versa would not work.

In addition, tuns can now properly be created in the target jail, as before the devfs rules were not being correctly reloaded.

FreeNAS Ticket: #40872

### CODE REVIEW

File: iocage_lib/ioc_common.py [hunk 1]
```diff
@@ -654,72 +654,68 @@ def check_release_newer(release, callback=None, silent=False):
...
+        # In case a plugin or something tries to use the human readable names
+        if include == '$devfsrules_hide_all':
+            include = '1'
+        elif include == '$devfsrules_unhide_basic':
+            include = '2'
+        elif include == '$devfsrules_unhide_login':
+            include = '3'
+
+        su.run(
+            ['devfs', 'rule', '-s', ruleset, 'add', 'include', include],
```
Comment (william-gr) [inline iocage_lib/ioc_common.py]: Hmm, why do we need to operate on /dev, and not on the devfs insided the jail?

Comment (skarekrow) [reply iocage_lib/ioc_common.py]: While we could operate on the jails /dev, users are able to specify their own rulesets. In which case those won't exist on the jail's /dev, and we can also get the devfs rules set up earlier then starting the jail by using the hosts. I don't see an advantage to using the jail's /dev.

**Comment by skarekrow** [comment]
labelling WIP as we discussed offline that we want to try a different route first, less invasive to the host.

File: iocage_lib/ioc_common.py [hunk 2]
```diff
@@ -654,58 +654,118 @@ def check_release_newer(release, callback=None, silent=False):
...
+        },
+            _callback=callback,
+            silent=silent)
+
+    ruleset = 5  # 0-4 is always reserved
+    while ruleset in ruleset_list:
+        ruleset += 1
+    ruleset = str(ruleset)
+
     devfs_dict = {
```
[the remark this answers is not in the document]
Comment (skarekrow) [reply iocage_lib/ioc_common.py]: I like it!

**Pull request merged**